expertsystem/data.py

```diff
@@ -63,11 +63,20 @@ class Spin(abc.Hashable):
     def __init__(self, magnitude: float, projection: float) -> None:
         magnitude = float(magnitude)
         projection = float(projection)
+        if magnitude % 0.5 != 0.0:
+            raise ValueError(
+                f"Spin magnitude {magnitude} has to be a multitude of 0.5"
+            )
         if abs(projection) > magnitude:
             raise ValueError(
                 "Spin projection cannot be larger than its magnitude:\n"
                 f"  {projection} > {magnitude}"
             )
+        if (projection - magnitude) % 1.0 != 0.0:
```

Comment:
I like the simplicity of this check. You could also consider using `is_integer`

Reply:
Ah yeah good one